src/app.ts

```diff
@@ -85,8 +88,41 @@ client.on('interactionCreate', async interaction => {
         server = await JoinGuild(interaction.guild)
     }
     await server.EnsureHasRole(interaction.member as GuildMember)
-
-    await ProcessCommand(server, interaction)
+    if(interaction.isCommand()) {
+        await ProcessCommand(server, interaction)
+    }
+    if(interaction.isButton()) {
+        //first 5 characters is the type of interaction, remaining is the name of the queue channel
+        const type = interaction.customId.substring(0,5)
+        const queue_name = interaction.customId.substring(5)
+        //console.log(queue_name + ' app.ts')
+        //console.log(interaction.member?.toString())
+        if(!(interaction.member instanceof GuildMember)) {
+            await interaction.reply({content: `Erm. Somethings wrong, this shouldn't happen. I'll inform the humaniod that maintains me`, ephemeral: true})
+            console.error(`Recieved an interaction without a member from user ${interaction.user} on server ${interaction.guild}`)
```

Comment:
An interaction without a member would mean that the bot received an interaction via direct message which shouldn't be possible. Just FYI.

Reply:
Ah ok, I get it now. The reason I had included this code was because it was giving me an error since I hadn't checked if interaction.member is an instance of GuildMember. So I just copied these lines from another part of the code.